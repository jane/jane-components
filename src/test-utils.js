// @flow

import * as React from 'react'
import renderer from 'react-test-renderer'
import theme from './theme'
import { ThemeProvider } from 'styled-components'

export const create = (C: React.Node): string =>
  renderer.create(<ThemeProvider theme={theme}>{C}</ThemeProvider>).toJSON()
