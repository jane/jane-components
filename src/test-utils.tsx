import * as React from 'react'
import renderer from 'react-test-renderer'
import theme from './theme'
import { ThemeProvider } from 'styled-components'

type RendererJson = {
  type: string
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  props: { [propName: string]: any }
  children: null | Array<RendererJson>
}

export const create = (C: React.ReactElement): RendererJson | null =>
  renderer.create(<ThemeProvider theme={theme}>{C}</ThemeProvider>).toJSON()
