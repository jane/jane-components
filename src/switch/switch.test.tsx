/* eslint-env jest */

import * as React from 'react'
import { mount } from 'enzyme'
import { create } from '../test-utils'
import Switch from './switch'

describe('switch', (): void => {
  const handleClick = jest.fn()
  const input = {
    value: false,
    onChange: handleClick,
  }
  const inputTrue = {
    value: true,
    onChange: handleClick,
  }

  it('renders', (): void => {
    expect(create(<Switch input={input} />)).toMatchSnapshot()
  })
  it('renders differently when checked', (): void => {
    expect(create(<Switch input={inputTrue} checked />)).toMatchSnapshot()
  })
  it('handles onChange on the input to toggle on', (): void => {
    const sc = mount(<Switch input={input} checked={false} />)
    expect(typeof sc.prop('input').onChange).toBe('function')
    sc.find('input').simulate('change')
    expect(sc.props().input.onChange).toHaveBeenCalled()
  })
  it('handles onChange on the input to toggle off', (): void => {
    const inputTrue = {
      value: true,
      onChange: handleClick,
    }
    const sc = mount(<Switch input={inputTrue} checked />)
    expect(typeof sc.prop('input').onChange).toBe('function')
    sc.find('input').simulate('click')
    expect(sc.props().input.onChange).toHaveBeenCalled()
  })
})
