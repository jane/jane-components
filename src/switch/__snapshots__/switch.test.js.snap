// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`switch renders 1`] = `
.c0 {
  background-color: #E0E0E0;
  border-radius: 24px;
  color: white;
  height: 33px;
  min-width: 60px;
  opacity: 1;
  position: relative;
  -webkit-transition: background-color 300ms cubic-bezier(0.215,0.61,0.355,1);
  transition: background-color 300ms cubic-bezier(0.215,0.61,0.355,1);
  width: 60px;
  opacity: 40%;
}

.c0:hover {
  background-color: #cccccc;
}

.c2 {
  background-color: white;
  border-radius: 24px;
  cursor: pointer;
  height: 25px;
  left: 4px;
  position: absolute;
  top: 4px;
  -webkit-transition: all 200ms cubic-bezier(0.215,0.61,0.355,1);
  transition: all 200ms cubic-bezier(0.215,0.61,0.355,1);
  width: 25px;
}

.c1 {
  cursor: pointer;
  height: 100%;
  opacity: 0;
  width: 100%;
}

<label
  checked={false}
  className="c0"
>
  <input
    aria-checked={false}
    checked={false}
    className="c1"
    onChange={[MockFunction]}
    type="checkbox"
    value={false}
  />
  <div
    checked={false}
    className="c2"
  />
</label>
`;

exports[`switch renders with disabled styles 1`] = `
.c0 {
  background-color: #E0E0E0;
  border-radius: 24px;
  color: white;
  height: 33px;
  min-width: 60px;
  opacity: .5;
  position: relative;
  -webkit-transition: background-color 300ms cubic-bezier(0.215,0.61,0.355,1);
  transition: background-color 300ms cubic-bezier(0.215,0.61,0.355,1);
  width: 60px;
}

.c0:hover {
  background-color: #cccccc;
}

.c2 {
  background-color: white;
  border-radius: 24px;
  cursor: not-allowed;
  height: 25px;
  left: calc(100% - 25px - 4px);
  position: absolute;
  top: 4px;
  -webkit-transition: all 200ms cubic-bezier(0.215,0.61,0.355,1);
  transition: all 200ms cubic-bezier(0.215,0.61,0.355,1);
  width: 25px;
}

.c1 {
  cursor: pointer;
  height: 100%;
  opacity: 0;
  width: 100%;
}

<label
  checked={true}
  className="c0"
  disabled={true}
>
  <input
    aria-checked={true}
    checked={true}
    className="c1"
    onChange={[MockFunction]}
    type="checkbox"
    value={true}
  />
  <div
    checked={true}
    className="c2"
    disabled={true}
  />
</label>
`;
